lib/commands/xctest-record-screen.js

```diff
@@ -60,8 +59,10 @@ async function retrieveRecodingFromSimulator(uuid) {
  * @returns {Promise<string>} The full path to the screen recording movie
  */
 async function retrieveRecodingFromRealDevice(uuid) {
-  const devicectl = new Devicectl(this.opts.udid, this.log);
-  const fileNames = await devicectl.listFiles(DOMAIN_TYPE, DOMAIN_IDENTIFIER, {
+  // @ts-expect-error - do not assign arbitrary properties to `this.opts`
+  const device = this.opts.device;
```

Comment:
it makes sense to explicitly provide the type of device, so typescript could verify further typechecks. Same note for other calls

Reply:
Yea, i wondered if it could. I need to learn more to add them as another pr. Initially I thought something like the below in driver.js, but maybe it was wrong.

```
/**
 * @typedef {import('@appium/types').DriverOpts<XCUITestDriverConstraints>} XCUITestDriverOpts
 * @property {import('./devicectl')Devicectl} device
 */
```

Reply:
I was thinking about having `device` property attached directly to the driver, like

```
get device () {
  return this.opts.device;
}
```

With proper typehints this show work nicely. Maybe I will take care bout it later myself after your changes are done